nixos/infrastructure/flyingcircus-virtual.nix

```diff
@@ -77,10 +78,34 @@ mkIf (cfg.infrastructureModule == "flyingcircus") {
       };
     };
 
-    services.serial-console-liveness = {
-      description = "Serial console liveness marker";
-      serviceConfig.Type = "oneshot";
-      script = "echo \"$(date) -- SERIAL CONSOLE IS LIVE --\" > /dev/ttyS0";
+    services = {
+      serial-console-liveness = {
+        description = "Serial console liveness marker";
+        serviceConfig.Type = "oneshot";
+        script = "echo \"$(date) -- SERIAL CONSOLE IS LIVE --\" > /dev/ttyS0";
+      };
+
+      fc-agent.serviceConfig =
+        let
+          # Must not consume more than 75% of available IOPS.
+          # The systemd setting is split between read and write but
+          # we only have one IOPS limit for the VM so combined we
+          # could get requests that are over the VM limit.
+          vdaIopsMax = "/dev/vda ${toString (maxIops - maxIops / 4)}";
+        in {
+          IOReadIOPSMax = vdaIopsMax;
+          IOWriteIOPSMax = vdaIopsMax;
+        };
+
+      fc-collect-garbage.serviceConfig =
+        let
+          # Must not consume more than 12.5% of available IOPS.
+          # We don't have a problem with this taking really long.
+          vdaIopsMax = "/dev/vda ${toString (maxIops / 8)}";
```

Comment:
can we make the comment and code a bit more aligned? I'd prefer `maxIops * 0.125` here ... ?

Reply:
Couldn't find a simple way to do multiplication and get an integer for the config string. Division yields an integer so that was the easiest way to do it.